### ISSUE TITLE
Proxy authentication

### ISSUE THREAD

**Issue by muzzar78** [OneGet/oneget#141]
I have been testing using the April preview on a corporate network and getting errors. Looking at the network dump the proxy server is returning authentication required. Is there anyway to specify to use default credentials for the proxy?

![oneget_proxy](https://cloud.githubusercontent.com/assets/13233154/8844183/780ea73c-3155-11e5-8ce4-947a5e219e21.png)

**Comment by ampmmn** [comment]
Hello,
I also ran into the above problem, using WMF 5.0 Production Preview on Windows 7.

So I looked into the source code as of @c00694c.
It seems not to set proxy parameter in Providers/Inbox/CoreProviders/WebDownloader/WebDownloader.cs.
The following is the code(WebDownloader.cs, WebDownloader.DownloadFile function).

```
    var webClient = new WebClient();

    webClient.Headers.Add("user-agent", "chocolatey command line");

    var done = new ManualResetEvent(false);

    webClient.DownloadFileCompleted += (sender, args) => {
        if (args.Cancelled || args.Error != null) {
            localFilename = null;
        }
        done.Set();
    };

    var lastPercent = 0;

    if (showProgress) {
        webClient.DownloadProgressChanged += (sender, args) => {
            // Progress(requestObject, 2, (int)percent, "Downloading {0} of {1} bytes", args.BytesReceived, args.TotalBytesToReceive);
            var percent = (int)((args.BytesReceived*100)/args.TotalBytesToReceive);
            if (percent > lastPercent) {
                lastPercent = percent;
                request.Progress(pid, (int)((args.BytesReceived*100)/args.TotalBytesToReceive), "To {0}", localFilename);
            }
        };
    }

    // start the download 
    webClient.DownloadFileAsync(remoteLocation, localFilename);

    ...
```

I think that webClient object should be set proxy parameter like this:

```
   var webClient = new WebClient();
   webClient.Proxy = System.Net.WebRequest.GetSystemWebProxy();
   webClient.Proxy.Credentials = CredentialCache.DefaultCredentials;
```

Could you prease address this issue?

Thanks.

**Comment by quoctruong** [comment]
Hi ampmmn, if you simply set "webClient.UseDefaultCredentials = true" instead of setting the proxy, will it work?

**Comment by jianyunt** [comment]
@muzzar78, this issue should be fixed now with the latest nugetprovider 2.8.5.202. Please try it out and let us know if your scenario works with the latest bits. thx.
